### PR TITLE
Implementing test on PR and cache GO dependencies.

### PR DESCRIPTION
**Pull request by petroskalligeris**

### CODE REVIEW

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # fetch tags
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
```
Comment (blewater) [inline .github/workflows/pr-test.yml:20]: let's include the `release` action update to 1.20 to this PR.

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: PR-Test
```
Comment (blewater) [inline .github/workflows/pr-test.yml:1]: please include an action with the [golangci/golangci-lint](https://github.com/golangci/golangci-lint) linter.

**Review by blewater — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-test.yml (hunk 1 above)
Comment (petroskalligeris) [reply .github/workflows/pr-test.yml:20]: @blewater  which release of go 1.20 do you want?

File: .github/workflows/pr-test.yml (hunk 2 above)
Comment (petroskalligeris) [reply .github/workflows/pr-test.yml:1]: added

File: .github/workflows/pr-test.yml (hunk 1 above)
Comment (blewater) [reply .github/workflows/pr-test.yml:20]: latest 1.20

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
```
Comment (blewater) [inline .github/workflows/golangci-lint.yml]: 1.20

File: .github/workflows/golangci-lint.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          cache: false
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # Require: The version of golangci-lint to use.
+          # When `install-mode` is `binary` (default) the value can be v1.2 or v1.2.3 or `latest` to use the latest version.
+          # When `install-mode` is `goinstall` the value can be v1.2.3, `latest`, or the hash of a commit.
```
Comment (blewater) [inline .github/workflows/golangci-lint.yml:30]: latest

File: .github/workflows/pr-test.yml (hunk 1 above)
Comment (petroskalligeris) [reply .github/workflows/pr-test.yml:20]: @blewater  you mean this one? goreleaser/goreleaser-action@master

File: .github/workflows/golangci-lint.yml (hunk 4 above)
Comment (petroskalligeris) [reply .github/workflows/golangci-lint.yml:30]: done

File: .github/workflows/golangci-lint.yml (hunk 3 above)
Comment (petroskalligeris) [reply .github/workflows/golangci-lint.yml]: done

File: .github/workflows/pr-test.yml (hunk 1 above)
Comment (petroskalligeris) [reply .github/workflows/pr-test.yml:20]: or you mean to make release.yml to run as well when open PR?

Comment (blewater) [reply .github/workflows/pr-test.yml:20]: release.yml to have its Go version also updated to 1.20. But that's ok I can do that. Let's wrap up this one.

**Review by blewater — APPROVED** [review]
Thanks

File: .github/workflows/pr-test.yml (hunk 1 above)
Comment (petroskalligeris) [reply .github/workflows/pr-test.yml:20]: ok, so this is done? I saw that it is already on 1.20

Comment (blewater) [reply .github/workflows/pr-test.yml:20]: Yeah, it's done. Thanks.

**Pull request merged**